x-pack/platform/plugins/shared/inference/server/chat_complete/anonymization/get_anonymizable_message_parts.ts

```diff
@@ -33,7 +34,21 @@ export function getAnonymizableMessageParts(message: Message) {
     };
   }
 
+  const content = message.content;
+  if (typeof content === 'string') {
+    return {
+      content,
+    };
+  } else if (Array.isArray(content)) {
+    return {
+      content: content
+        // only return text content and don't return image content
+        .filter((item): item is MessageContentText => item.type === 'text')
+        .map((item) => ({ text: item.text })),
+    };
+  }
```

Comment:
nit: We do not need three branches. Two should be enough:

```suggestion
  const content = message.content;
  if (Array.isArray(content)) {
    return {
      content: content
        // only return text content and don't return image content
        .filter((item): item is MessageContentText => item.type === 'text')
        .map((item) => ({ text: item.text })),
    };
  }

  return { content };
```